lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -314,367 +316,417 @@ def concatenate_cubes(
+        Args:
+            cubes (iris.cube.CubeList or iris.cube.Cube):
+                Cubes to be merged.
 
-    if len(unmatching_attributes) > 0:
-        for i, cube in enumerate(cubes):
-            # Remove ignored attributes.
-            for attr in silent_attributes:
-                if attr in unmatching_attributes[i]:
-                    cube.attributes.pop(attr)
-                    unmatching_attributes[i].pop(attr)
-
-            # If a model_id_attr has been specified, assume we are trying to
-            # grid blend and throw an error if the model_id_attr does not
-            # match that on the cubes to be blended.
-            if model_id_attr is not None and \
-                    model_id_attr not in cube.attributes:
-                msg = ('Cannot create model ID coordinate for grid blending '
-                       'as the model ID attribute specified is not found '
-                       'within the cube attributes')
-                raise ValueError(msg)
+        Kwargs:
+            check_time_bounds_ranges (bool):
+                Flag to check whether scalar time bounds ranges match.
+                This is for when we are expecting to create a new "time" axis
+                through merging for eg precipitation accumulations, where we
+                want to make sure that the bounds match so that we are not eg
+                combining 1 hour with 3 hour accumulations.
 
-            if model_id_attr in unmatching_attributes[i]:
-                model_title = cube.attributes.pop(model_id_attr)
-                cube.attributes[model_id_attr] = "blend"
-
-                new_model_id_coord = build_coordinate([1000 * i],
-                                                      long_name='model_id',
-                                                      data_type=np.int)
-                new_model_coord = (
-                    build_coordinate([model_title],
-                                     long_name='model_configuration',
-                                     coord_type=AuxCoord,
-                                     data_type=np.str))
-
-                cube.add_aux_coord(new_model_id_coord)
-                cube.add_aux_coord(new_model_coord)
-                unmatching_attributes[i].pop(model_id_attr)
-
-            # Remove any other mismatching attributes but raise warning.
-            if len(unmatching_attributes[i]) != 0:
-                for key in unmatching_attributes[i]:
-                    msg = ('Do not know what to do with ' + key +
-                           ' will delete it'
-                           ' - value is {}'.format(cube.attributes[key]))
-                    warnings.warn(msg)
-                    cube.attributes.pop(key)
-    return cubes
+        Returns:
+            iris.cube.Cube:
+                Merged cube.
+        """
+        # if input is already a single cube, return unchanged
+        if isinstance(cubes_in, iris.cube.Cube):
```

Comment:
or if len(cubelist) ==1? (relates to a comment somewhere else about unit tests for cubelists with only one cube in)

Reply:
Sorted (see above)